system-functionalities.md

```diff
@@ -338,14 +338,40 @@ Implementation guidance: For security, all verification checks run by the key se
 The asset owner can create and use signing keys in a manner similar to arbitrary secrets, with the additional operation of creating signatures.
 
 ### Inherited Operations
-All of the [protocols for arbitrary secrets](#operations-on-arbitrary-secrets) should be supported. The only difference is the type of key created and used. That is, calling the generation functionality should allow the asset owner to create one of the following key types:
+All of the [protocols for arbitrary secrets](#operations-on-arbitrary-secrets) should be supported. The only significant difference is the type of key created and used; minor protocol modifications are described below. That is, calling the generation functionality should allow the asset owner to create one of the following key types:
 - ECDSA on secp256 curve; or
 - Ed25519 (i.e., EdDSA on edwards25519).
 
-Signing keys have an additional supported operation, namely, the creation of a signature.
+Signing keys have an additional supported operation, namely, the creation of a signature. They also have an additional public component that can be derived from a generated key.
 
 Implementation guidance: We pick these ECDSA/EdDSA parameters in order to provide functionality compatible with EVM-based blockchains. However, we expect to support multiple blockchains and signing primitives in the future.
 
+### Generating and Storing a Signing Key
+Both key generation protocols need a modification to return the public component of the generated key to the calling application.
+
+In [local secret generation with remote backup](#local-secret-generation-with-remote-encrypted-backup):
+- In step [3.iii](#localgen-client-generate), the generate protocol returns the `arbitrary_key` and its public component `public_key`.
+- In step [5.i](#localgen-client-output), the client outputs `key_id`, `public_key`, and `StorageObject` to the calling application.
+
+In [remote-only secret generation protocol](#remote-only-secret-generation-and-storage):
+- In step [2.iii](#remotegen-server-generate), the generate protocol returns the `arbitrary_key` and its public component `public_key`.
+- In step [2.v](#remotegen-server-send-key-id), the key server sends `key_id` and `public_key` to the client over the secure channel.
+- In step [3.i](#remotegen-client-store), the client stores `key_id` and `public_key`.
```

Comment:
Maybe we are at a point where we need to figure a better way of talking about local storage. There is still tension in the spec with respect to this point: are we describing a library or a product built on the library.

Reply:
Yeah, I agree. If we are aiming for the spec to describe the functionality of the code, I think it's better to use terminology like "the client returns X and Y to the calling application to store".

Reply:
Right, I think we've been struggling because we don't have anything concrete about how local storage should work and how to integrate with our code. I think we will develop guidance around local storage (and possibly a utility of some kind?) down the line, and that should make it obvious where/how to integrate.